Add a doc page for ocean diagnostics - diag_table (#1020)
COSIMA/MOM6-CICE6: documentation/docs/pages/configurations/Ocean_diagnostics.md

The generated `diag_table` is treated as a derived file and should generally not be edited by hand, as manual changes may be overwritten when configurations are regenerated.

## `diag_table` Structure
The `diag_table` consists of three sections: `<title>`, `<file>`, and `<field>`.

- The title section is mandatory. It appears at the top of the file and consists of a title string and a reference date specified by 6 integers.
- The file section defines diagnostic output files and their output frequency. Each file entry may contain one or more diagnostic fields.
- The field section defines individual diagnostic fields. Because MOM6 is a general-coordinate model, diagnostics may be written either in the native model coordinate system, or in user-defined (remapped) coordinates.

## ACCESS-OM3 diagnostic filename conventions
The diagnostic filename conventions used by ACCESS-OM3 configurations are summarised [here](https://github.com/ACCESS-NRI/access-om3-configs/issues/374). At a high level, diagnostic files follow the pattern:
```
<file_prefix>.<model>.<dimension_or_mode>.<field_or_mode>[.<vertical_coordinate>][.<d2>].<frequency>.<time_cell_method>.<datestamp>.nc
```

> !!! important

Author: ```suggestion
!!! tip
```

# Ocean diagnostics

Ocean diagnostics in MOM6 are configured via the `diag_table` file, which controls all runtime diagnostic output. Full details of the `diag_table` format and semantics are available in the [MOM6 docs](https://mom6.readthedocs.io/en/main/api/generated/pages/Diagnostics.html). 

> In ACCESS-OM3, `diag_table` is treated as a derived file and should not be edited directly, as any manual changes may be overwritten when configurations are regenerated. Instead, it is generated from the user-editable configuration file `diag_table_source.yaml` using the Python helper script [`make_diag_table.py`](https://github.com/COSIMA/make_diag_table), , which is available under `/g/data/vk83/apps/make_diag_table/make_diag_table.py`. Hence this documentation mainly focuses on the resulting diagnostic filename conventions, rather than the generation workflow.

Reviewer: ```suggestion
In ACCESS-OM3, `diag_table` is treated as a derived file and should not be edited directly, as any manual changes may be overwritten when configurations are regenerated. Instead, it is generated from the user-editable configuration file `diag_table_source.yaml` using the Python helper script [`make_diag_table.py`](https://github.com/COSIMA/make_diag_table), which is available under `/g/data/vk83/apps/make_diag_table/make_diag_table.py`. Hence this documentation mainly focuses on the resulting diagnostic filename conventions, rather than the generation workflow.
```

## ACCESS-OM3 diagnostic filename conventions

> The diagnostic filename conventions used by ACCESS-OM3 configurations are summarised [here](https://github.com/ACCESS-NRI/access-om3-configs/issues/374) and [here](https://github.com/ACCESS-NRI/access-om3-configs/issues/1024). At a high level, diagnostic files follow the pattern:

Reviewer: ```suggestion
The diagnostic filename conventions used by ACCESS-OM3 configurations are summarised [here](https://github.com/ACCESS-NRI/access-om3-configs/issues/374#issuecomment-2750096126). At a high level, diagnostic files follow the pattern:
```

```
<file_prefix>.<model>.<dimension>[.<field>[+<vertical_coordinate>][+<d2>]].<frequency>.<time_cell_method>.<datestamp>.nc
```

> !!! important

Reviewer: ```suggestion
!!! note
```

>     In practice, filenames fall into a small number of well-defined cases, described below.

Reviewer 2: > I’m on macOS. [@chrisb13](https://github.com/chrisb13), are you on Windows?

Yeah, chrome.

I'm on a different (work) computer this morning and it looks ok now, not sure if that's changing to `note` though.

>     In practice, filenames fall into a small number of well-defined cases, described below.

Reviewer 3: There is no `!!! important` admonition.

    - For special cases:
        - `static` files do not include a field name,
        - `scalar` files do not include a field name and instead contain multiple scalar diagnostics in a single file.

!!! note
    The ACCESS-OM3 convention is to write one physical field per file, except in the two special cases above.

### Temporal components
- `frequency`: The output interval between records, following naming conventions given in [this schema](https://github.com/ACCESS-NRI/schema/blob/dfaf21913f1159a4d76b70ea783a0ecb54798631/au.org.access-nri/model/output/file-metadata/1-0-1.json#L68-L92).
- `<time_cell_method>`: Specifies how values are accumulated within each output interval, such as "mean", "max", "min", "snap" etc.
- `<datestamp>`: Uses FMS time-string formatting, such as "%4yr" (4-digit year), other options can be "%dy" (day), "%mo" (month), "%hr" (hour), "%mn" (minute), "%sc" ("second") etc. The datestamp indicates the frequency at which new files are written. E.g. `YYYY` indicates yearly, `YYYY-MM` indicates monthly etc.

### Optional spatial components
- `<vertical_coordinate>`: Is included only for non-native vertical coordinates, such as `z`, `rho2` etc

> - `<d2>`: Indicates half-resolution (downsampling) diagnostics. This is only supported for standard `2d` / `3d` spatial diagnostics and requires:

Reviewer: ```suggestion
- `d2`: Indicates half-resolution (downsampling) diagnostics. This is only supported for standard `2d` / `3d` spatial diagnostics and requires:
```

    - `NIGLOBAL` / `Layout_X` divisible by 2
    - `NJGLOBAL` / `Layout_Y` divisible by 2

    where `NIGLOBAL` and `NJGLOBAL` are the global horizontal grid sizes, and `Layout_X`, `Layout_Y` are the processor layouts. Related issue was discussed [here](https://github.com/ACCESS-NRI/access-om3-configs/issues/539).

### Practical filename cases

> In practice, ACCESS-OM3 uses three distinct filename classes.

Reviewer: ```suggestion
In practice, with the above conventions ACCESS-OM3 uses three distinct filename classes.
```

## `diag_table` Structure
The `diag_table` consists of three sections: `<title>`, `<file>`, and `<field>`.

- The title section is mandatory. It appears at the top of the file and consists of a title string and a reference date specified by 6 integers.
- The file section defines diagnostic output files and their output frequency. Each file entry may contain one or more diagnostic fields.
- The field section defines individual diagnostic fields. Because MOM6 is a general-coordinate model, diagnostics may be written either in the native model coordinate system, or in user-defined (remapped) coordinates.

## ACCESS file name naming schemes
The diagnostic filename conventions used by ACCESS-OM3 configurations are summarised [here](https://github.com/ACCESS-NRI/access-om3-configs/issues/374#issuecomment-2750096126). At a high level, diagnostic files follow the pattern:
```
<file_prefix>.<model>.<dimension_or_mode>.<field_or_mode>[.<vertical_coordinate>][.<d2>].<frequency>.<time_cell_method>.<datestamp>.nc
```

> !!! tip Although several components are shown as optional, not all combinations are valid. In practice, filenames fall into a small number of well-defined cases, described below.

Author: I thought it would render to be different. Any suggestions on this @atteggiani ?

> !!! tip Although several components are shown as optional, not all combinations are valid. In practice, filenames fall into a small number of well-defined cases, described below.

Reviewer 2: ```suggestion
!!! tip
    Although several components are shown as optional, not all combinations are valid. In practice, filenames fall into a small number of well-defined cases, described below.
```

Author: It works! Thanks @chrisb13

Reviewer 3: The icons used in the Hive Docs are not present in this repo. They should be added. 
Please add [fontawesome CDN](https://github.com/ACCESS-NRI/ACCESS-Hive-Docs/blob/0b13b4a376bf1bd033d8e6d73e0281911b89bc1c/mkdocs.yml#L227) to the `mkdocs.yml`.
